src/main/java/de/siegmar/logbackgelf/SimpleJsonEncoder.java

```diff
@@ -107,36 +107,44 @@ private void appendKey(final String key) throws IOException {
      */
     @SuppressWarnings("checkstyle:cyclomaticcomplexity")
     private void escapeString(final String str) throws IOException {
+        int currentIndex = 0;
+        int lastCopyIndex = 0;
         for (final char ch : str.toCharArray()) {
+            String escaped = null;
             switch (ch) {
                 case QUOTE:
                 case '\\':
                 case '/':
-                    writer.append('\\');
-                    writer.append(ch);
+                    escaped = "\\" + ch;
                     break;
                 case '\b':
-                    writer.append("\\b");
+                    escaped = "\\b";
                     break;
                 case '\f':
-                    writer.append("\\f");
+                    escaped = "\\f";
                     break;
                 case '\n':
-                    writer.append("\\n");
+                    escaped = "\\n";
                     break;
                 case '\r':
-                    writer.append("\\r");
+                    escaped = "\\r";
                     break;
                 case '\t':
-                    writer.append("\\t");
+                    escaped = "\\t";
                     break;
-                default:
-                    if (ch < ' ') {
-                        writer.append(escapeCharacter(ch));
-                    } else {
-                        writer.append(ch);
-                    }
             }
+            if (ch < ' ') {
```

Comment:
Why don't you keep this if in the switch (default) block?

Reply:
The code looked simpler that way, but I didn't realize that it would add an unnecessary if condition check when the current char is an escaped char. Moved the if back to the default block